scaling_performance/optimizing_storage.adoc

```diff
@@ -11,19 +11,167 @@
+
+| File
+| This type of storage is presented to the operating system as a file system export to be mounted. File storage is
+sometimes also referred to as Network Attached Storage (NAS). Concurrency, latency, file locking mechanisms and other
+capabilities will vary widely between protocols, implementations, vendors, and scales.
+| CNS/CRS GlusterFS footnoteref:[disclaimer], RHEL NFS, NetApp NFS footnoteref:[disclaimer2, NetApp NFS supports Dynamic PV provisioning when using the Trident plugin.], Azure File, Vendor NFS, Vendor GlusterFS footnoteref:[disclaimer3, Vendor GlusterFS, Vendor S3, and Vendor Swift supportability and configurability may vary.], Azure File, AWS EFS
+
+| Object
+| This type of is storage is accessible via a REST API endpoint. Object storage is configurable for use in the {product-title} Registry. Applications
+intending to use this storage must build their drivers into the application and/or container.
+| CNS/CRS GlusterFS footnoteref:[disclaimer], Ceph Object Storage (RADOS Gateway), OpenStack Swift, Aliyun OSS, AWS S3, Google Cloud Storage, Azure Blob Storage, Vendor S3 footnoteref:[disclaimer3], Vendor Swift footnoteref:[disclaimer3]
+|===
+
+[NOTE]
+====
+As of {product-title} 3.6.1, Container-Native Storage (CNS) GlusterFS (a hyperconverged or cluster-hosted storage solution) and Container-Ready Storage (CRS)
+GlusterFS (an externally hosted storage solution) provides interfaces for block, file, and object storage for the purpose of the {product-title} registry, logging, and metrics.
+====
+
+[[back-end-recommendations]]
+=== Storage Back-end Recommendations
+
+The following table summarizes the recommended and configurable storage back-end types for the given {product-title} cluster application.
+
+.Recommended and configurable storage back-end types
+[options="header"]
+|===
+|Storage type |ROX |RWX |Registry|Scaled registry| Metrics|Logging|Apps
+
+| Block
+| Yes footnoteref:[disclaimer4, This does not apply to physical disk, VM physical disk, VMDK, loopback over NFS, AWS EBS, and Azure Disk.]
+| No
+| Configurable
+| Not configurable
+| Recommended
+| Recommended
+| Recommended
+
+| File
+| Yes footnoteref:[disclaimer4]
+| Yes
+| Configurable
+| Configurable
+| Configurable
+| Configurable
+| Recommended
+
+| Object
+| Yes
+| Yes
+| Recommended
+| Recommended
+| Not configurable
+| Not configurable
+| Not configurable footnoteref:[disclaimer5, Object storage is not consumed via {product-title}'s PVs/PVCs. Apps must integrate with the object storage REST API.]
+|===
+
+[NOTE]
+====
+A scaled registry is an {product-title} registry where three or more pod replicas are running.
+====
+
+[[application-storage-recommendations]]
+==== Specific Application and Storage Recommendations
+
+[[registry]]
+===== Registry
+
+In a non-scaled/HA {product-title} registry cluster deployment, the preferred storage back-end type is object storage followed by block storage. The
```

Comment:
For this section, I think we should use points for each sentence, like:
```
In a non-scaled/HA {product-title} registry cluster deployment:

* The preferred storage back-end type is object storage followed by block
storage.  The storage back-end type does not need to support ReadWriteMany
* (RWX) access mode.  The storage back-end type must ensure read-after-write
* consistency.  All NAS back-end types (excluding CNS/CRS GlusterFS as it uses
an object storage interface) are expressly not recommended for {product-title}
Registry cluster deployment with production workloads.  
* While hostPath volumes are configurable for a non-scaled/HA {product-title}
Registry cluster deployment, they are not recommended.
```

Reply:
Same for Scaled registry, metrics, logging, and applications section if possible.

Reply:
Should we also highlight whats not recommend by using admonitions? I am not sure. :confused: :grey_question:

Reply:
@gaurav-nelson YES!  We do need to mention what to use. Here is an email from openshift-users email list from yesterday, 11/1/2017.

```
I am confused over persistent storage for logging (elasticsearch).

The latest advanced installer docs [1] specifically describes how to define using NFS for persistent storage, but the docs for "aggregating container logs" [2] says that NFS should not be used (except in one particular scenario) and seems to suggest that the only really suitable scenario is to use a volume (disk) directly mounted to each logging node.

Could someone clarify the situtation?
```

My goal is to reference this page from the Registry, Metrics, and Logging install docs once this PR lands.  There are probably a couple other places which could do with a reference to this doc as well.